src/util.c

```diff
@@ -81,6 +82,44 @@ GladeXML *gtt_glade_xml_new(const char *filename, const char *widget)
     return xml;
 }
 
+static GtkBuilder *gtt_builder_new_from_exact_file(const char *filename)
+{
+    GtkBuilder *builder = NULL;
+    GError *err = NULL;
+
+    builder = gtk_builder_new();
+
+    gtk_builder_add_from_file(builder, filename, &err);
+
+    // For now, let's just die. Don't forget g_error_free()
+    // in case this is changed.
+    if (err != NULL)
+        g_error("Error loading UI: %s", err->message);
+
```

Comment:
Not entirely sure what you mean here. `gtk_builder_new_from_file` (available from 3.10) does pretty much the [same thing](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gtk/gtkbuilder.c#L2873) - i.e. it dies right here. It does include the filename in the error message, which I forgot to do, so I'll fix that.

I feel like that should be pretty likely to give a good indication of what the problem is. Returning NULL would either force all callers to explicitly check for it and output a good message themselves, or if that check is forgotten I suppose it would seg fault which is not likely to output a decent error message.

Btw, later I suppose it would be preferable to switch to `gtk_builder_new_from_resource` to avoid the matter of file location.

Reply:
I improved the message to contain the filename, but otherwise left the logic as is.

Reply:
Sorry, my double-bad. I didn't notice `gtk_builder_new_from_file` is available from 3.10 only and didn't think of the effect of `g_error`. It looks good, thank you :-)